gtsfm/densify/mvs_base.py

```diff
@@ -0,0 +1,62 @@
+"""Base class for the MVS stage of the back-end.
+
+Authors: John Lambert, Ren Liu
+"""
+import abc
+from typing import Dict, Tuple
+
+import dask
+import numpy as np
+from dask.delayed import Delayed
+from gtsam import PinholeCameraCal3Bundler
+
+from gtsfm.common.image import Image
+
+
+class MvsBase(metaclass=abc.ABCMeta):
+    """Base class for all multi-view stereo hyperparameters."""
+
+    def __init__(self) -> None:
+        """Initialize the MVS module """
+        pass
+
+    @abc.abstractmethod
+    def densify(
+        self,
+        images: Dict[int,Image],
+        cameras: Dict[int, PinholeCameraCal3Bundler],
+        min_distance: float,
+        max_distance: float,
+    ) -> Tuple[np.ndarray, Dict[int, np.ndarray]]:
+        """Densify a point cloud using multi-view stereo
+
+        Args:
+            image: dictionary mapping image indices to input images.
+            cameras: dictionary mapping image indices to camera parameters
+            min_distance: minimum distance from any camera to any 3d point
+            max_distance: maximum distance from any camera to any 3d point
```

Comment:
Is it depth or distance? If it is used to determine some cost volume in disparity space, it might be depth?

Reply:
It should be the depth from any camera to the detect points in the sparse point cloud, which is in the same scale as points coordinates.

Reply:
John, I understand distance to be norm of the 3D position in camera frame and depth to be the distance along the optical axis. They’re not equivalent

Reply:
Good point, thanks for catching that Akshay